### ISSUE TITLE
Vercel deploy error:  Cannot read properties of undefined (reading 'vendor')

### ISSUE THREAD

**Issue by altrome** [jorge-menjivar/unSAGED#13]
Trying to deploy on vercel but getting the following error on browser:

Application error: a client-side exception has occurred (see the browser console for more information).

the console has the following erors:
> Section personalization not found
> Cannot read properties of undefined (reading 'vendor')

This last error seems to be related to model... 

env vars defined (Checked correct values):
GOOGLE_CLIENT_SECRET
GOOGLE_CLIENT_ID
NEXTAUTH_URL
NEXTAUTH_SECRET
NEXT_PUBLIC_SUPABASE_URL
NEXT_PUBLIC_SUPABASE_ANON_KEY
SUPABASE_SERVICE_ROLE_KEY
SUPABASE_JWT_SECRET

try adding  DEFAULT_MODE and still not working...

Any ideas how to solve this?

**Comment by jorge-menjivar** [comment]
Hi @altrome, thank you for raising this issue. 

I'm trying to reproduce this on Vercel with only those variables set but not being able to.

Can you provide some more information:
- Are you able to run on local computer?
- What steps did you take to deploy on vercel?
- Does this happen at startup or after login page?
- Is this error itself being thrown by NextJs or the browser-console? If so, are you able to see any more error logs?

**Comment by altrome** [comment]
Hi @jorge-menjivar, thanks for your amazing work!

I've only tested at Vercel, not on local. This error happens after login, when loads the chat and shown at browser-console. 

As far as I have been able to investigate, the errors seems to be initially located at SystemPromptSelect.tsx file. In particular, the methods getAvailableSystemPrompts &  getDefaultSystemPrompt, when trying to select the model of the selectedConversation. If you start from scratch, without any conversations, the model is undefined so you can not select the vendor later on the code. I've figured out a workaround, is still trowing other errors, but at least I can test your version.

In short, I believe that the errors thrown by the platform have to do with the fact that there are no conversations available and therefore you cannot select the model, and hence the errors are triggered.

**Comment by jorge-menjivar** [comment]
I was able to reproduce the described issue with a fresh account on a vercel instance. However, on local it was giving a different error. I fixed the error described in the local instance which also seems to have fixed the vercel issue. Let me know if you continue to have this issue and I will investigate further.
